pymc3/model.py

```diff
@@ -931,6 +954,20 @@ def cont_vars(self):
         """All the continuous variables in the model"""
         return list(typefilter(self.value_vars, continuous_types))
 
+    def next_rng(self) -> RandomStateSharedVariable:
+        """Generate a new ``RandomStateSharedVariable``.
+
+        The new ``RandomStateSharedVariable`` is also added to
+        ``Model.rng_seq``.
+        """
+        new_seed = self.rng_seeder.randint(2 ** 30, dtype=np.int64)
```

Comment:
With this approach, the same seed should still produce a distinct shared state object and, thus, not compare equal.

Reply:
Got it

Reply:
This example demonstrates how the same RNG states can be used by distinct shared RNG objects and preserve the identities of their corresponding `RandomVariable`s:
```python
import numpy as np

import aesara
import aesara.tensor as at


rng_state = aesara.shared(
    np.random.RandomState(np.random.MT19937(np.random.SeedSequence(1234)))
)

rng_state2 = aesara.shared(
    np.random.RandomState(np.random.MT19937(np.random.SeedSequence(1234)))
)


x = at.random.normal(rng=rng_state)
y = at.random.normal(rng=rng_state)
z = at.random.normal(rng=rng_state2)


fn = aesara.function([], x + y + z, mode="FAST_RUN")

aesara.dprint(fn.maker.fgraph)
# Elemwise{Add}[(0, 2)] [id A] ''   2
#  |normal_rv{0, (0, 0), floatX, False}.1 [id B] ''   1
#  | |RandomStateSharedVariable(<RandomState(MT19937) at 0x7F6CD2960E20>) [id C]
#  | |TensorConstant{[]} [id D]
#  | |TensorConstant{11} [id E]
#  | |TensorConstant{0.0} [id F]
#  | |TensorConstant{1.0} [id G]
#  |normal_rv{0, (0, 0), floatX, False}.1 [id B] ''   1
#  |normal_rv{0, (0, 0), floatX, False}.1 [id H] ''   0
#    |RandomStateSharedVariable(<RandomState(MT19937) at 0x7F6CD27BC050>) [id I]
#    |TensorConstant{[]} [id D]
#    |TensorConstant{11} [id E]
#    |TensorConstant{0.0} [id F]
#    |TensorConstant{1.0} [id G]

# Let's confirm that these all produce the same thing, because
# they have the same RNGs...
fn2 = aesara.function([], [x, y, z], mode="FAST_COMPILE")

fn2()
# [array(-0.24304841), array(-0.24304841), array(-0.24304841)]
```

Reply:
That's exactly what I was wondering. Great. 

And something like this wouldn't be useful having on Aesara test suit either?

Reply:
No, not really.